### PR TITLE
Copy over headers we use for firebase for the android distribution

### PR DESCRIPTION
**Pull request by hyp**
### Description
This changes the android firebase-cpp-sdk to also include the headers we use in the windows distribution in the android distribution as well.

***
### Testing
Manual workflow run.
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

### CODE REVIEW

**Comment by hyp** [comment]
Merging to try it out in the firebase CI / local builds.

File: .github/workflows/bcny-firebase.yml [hunk 1]
```diff
@@ -475,6 +475,29 @@ jobs:
...
+          cp ${firebase_dir}/auth/src/include/firebase/auth/*.h ${header_destination_dir}/auth/
+          cp ${firebase_dir}/firestore/src/include/firebase/firestore.h ${header_destination_dir}
+          mkdir -p $header_destination_dir/firestore
+          cp ${firebase_dir}/firestore/src/include/firebase/firestore/*.h ${header_destination_dir}/firestore/
+          cp ${firebase_dir}/functions/src/include/firebase/functions.h ${header_destination_dir}
+          mkdir -p $header_destination_dir/functions
+          cp ${firebase_dir}/functions/src/include/firebase/functions/*.h ${header_destination_dir}/functions/
+          cp ${firebase_dir}/storage/src/include/firebase/storage.h ${header_destination_dir}
+          mkdir -p $header_destination_dir/storage
+          cp ${firebase_dir}/storage/src/include/firebase/storage/*.h ${header_destination_dir}/storage/
```
Comment (compnerd) [inline .github/workflows/bcny-firebase.yml]: Can you change this stylistically?

```sh
mkdir -p ${header_destination_dir}/firestore ${header_destination_dir}/functions ...
for header in ... ; do
  cp ...
done
```

Comment (hyp) [reply .github/workflows/bcny-firebase.yml]: Sure!

**Comment by hyp** [comment]
Updated, merging

**Pull request merged**